packages/edit-site/src/components/sidebar-navigation-screen-pattern/template-part-navigation-menu-list-item.js

```diff
@@ -11,7 +13,20 @@ import SidebarNavigationItem from '../sidebar-navigation-item';
 import { useLink } from '../routes/link';
 
 export default function TemplatePartNavigationMenuListItem( { id } ) {
-	const [ title ] = useEntityProp( 'postType', 'wp_navigation', 'title', id );
+	const title = useSelect( ( select ) => {
+		const { getEditedEntityRecord } = select( coreStore );
+		const { __experimentalGetTemplateInfo: getTemplateInfo } =
+			select( editorStore );
+
+		const _record = getEditedEntityRecord(
+			'postType',
+			'wp_navigation',
+			id
+		);
+
+		const templateInfo = getTemplateInfo( _record );
+		return templateInfo?.title || __( '(no title)' );
+	} );
```

Comment:
Can we just use `record?.title?.rendered || record?.title` here? We can leave `no title` fallback outside of the `mapSelect`.

Using the selector `getTemplateInfo` meant for different entities can have unexpected results in the future.

P.S. The `useSelect` is missing a dependency here.